Algorithm.CSharp/CustomDataUsingMapFileRegressionAlgorithm.cs

```diff
@@ -0,0 +1,146 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom.SEC;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders.Fees;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm demonstrating use of map files with custom data
+    /// </summary>
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="custom data" />
+    /// <meta name="tag" content="regression test" />
+    /// <meta name="tag" content="SEC" />
+    /// <meta name="tag" content="rename event" />
+    /// <meta name="tag" content="map" />
+    /// <meta name="tag" content="mapping" />
+    /// <meta name="tag" content="map files" />
+    public class CustomDataUsingMapFileRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _symbol;
+        private bool _changedSymbol;
+        private bool _properSymbolBeforeRename;
+        private bool _properSymbolAfterRename;
+
+        /// <summary>
+        /// Ticker we use for testing
+        /// </summary>
+        public const string Ticker = "TWX";
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2001, 1, 1);
+            SetEndDate(2003, 12, 31);
+            SetCash(100000);
+
+            // AOL renames to TWX in 2003
+            _symbol = AddData<SECReport8K>(Ticker, Resolution.Daily).Symbol;
```

Comment:
I think it makes sense to add some of the other custom data types to this regression test so we have better coverage, especially since some of the other types are much older, and so they may behave differently than expected.

Reply:
I agree, but I don't think that any of the other data sources bundled with LEAN at the moment have support for mapping except SEC and Psychsignal (which we can't include)

Reply:
Very well, I suppose I was thinking we could add a quandl type that could be mapped or just define a new custom data type specifically for this regression algorithm that does the AOL->TWX mapping. It's always nice for users to be able to see how to use a feature if they need to implement their own.